src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -121,16 +121,16 @@ BoundExpression MakeDeclarationPattern(SyntaxNode syntax, BoundExpression input,
                 if (input.Type == type)
                 {
                     // CONSIDER: this can be done whenever input.Type is a subtype of type for improved code
-                    var assignment = _factory.AssignmentExpression(_factory.Local(target), input);
+                    var assignment = _factory.AssignmentExpression(target, input);
                     var result = requiresNullTest
-                        ? _factory.ObjectNotEqual(_factory.Local(target), _factory.Null(type))
+                        ? _factory.ObjectNotEqual(target, _factory.Null(type))
```

Comment:
>target [](start = 50, length = 6)

It looks like we are duplicating arbitrarily complex expression in the resulting tree, feels fragile in the long term.